pnpm@7.18.0 fails to install dependencies with absolute `file:/path/to/local/version` overrides - worked in 7.17.1

<!--
  If this issue affects many people in a company/big team, create a post for your company in the following discussion:
  https://github.com/pnpm/pnpm/discussions/3787
  and link the issue in your post.

  This will help us prioritize issues that affect more people.
-->

### pnpm version: 7.18.0

### Code to reproduce the issue:

create an override to a local checkout with absolute file path:
```json
{
  "devDependencies": {
    "a": "file:/absolute/path/to/a"
  },
  "pnpm": {
    "overrides": {
      "a": "file:/absolute/path/to/a"
    }
  }
}
```
https://github.com/dominikg/pnpm-718-override-repro

### Expected behavior:

installs local override

### Actual behavior:

errors with the absolute path appended to cwd
```
reproductions/pnpm-718-issue/b ❯ pnpm i
 ERR_PNPM_LINKED_PKG_DIR_NOT_FOUND  Could not install from "/home/dominikg/develop/reproductions/pnpm-718-issue/b/home/dominikg/develop/reproductions/pnpm-718-issue/a" as it does not exist.

```

### Additional information:

 - `node -v` prints: 18.11.0
 - Windows, macOS, or Linux?: linux
 
 this pattern used to work great in vite-ecosystem-ci to provide a local built version of vite to downstream project cis. 
 relative `file:../../` still work in pnpm 7.18 but are not sufficient as we have to be able to apply the override on different nesting levels of monorepo packages.

--- a/.changeset/perfect-islands-trade.md
+++ b/.changeset/perfect-islands-trade.md
@@ -1,0 +1,6 @@
+---
+"@pnpm/hooks.read-package-hook": patch
+"pnpm": patch
+---
+
+It should be possible to use overrides with absolute file paths [#5754](https://github.com/pnpm/pnpm/issues/5754).

--- a/hooks/read-package-hook/src/createVersionsOverrider.ts
+++ b/hooks/read-package-hook/src/createVersionsOverrider.ts
@@ -18,7 +18,8 @@ export function createVersionsOverrider (
         }
         let linkFileTarget: string | undefined
         if (override.newPref.startsWith('file:')) {
-          linkFileTarget = path.join(rootDir, override.newPref.substring(5))
+          const pkgPath = override.newPref.substring(5)
+          linkFileTarget = path.isAbsolute(pkgPath) ? pkgPath : path.join(rootDir, pkgPath)
         }
         return {
           ...override,

--- a/hooks/read-package-hook/test/createVersionOverrider.test.ts
+++ b/hooks/read-package-hook/test/createVersionOverrider.test.ts
@@ -271,3 +271,23 @@ test('createVersionsOverrider() overrides dependencies with file', () => {
     },
   })
 })
+
+test('createVersionsOverrider() overrides dependencies with file specified with absolute path', () => {
+  const absolutePath = path.join(__dirname, 'qar')
+  const overrider = createVersionsOverrider({
+    qar: `file:${absolutePath}`,
+  }, process.cwd())
+  expect(overrider({
+    name: 'foo',
+    version: '1.2.0',
+    dependencies: {
+      qar: '3.0.0',
+    },
+  }, path.resolve('pkg'))).toStrictEqual({
+    name: 'foo',
+    version: '1.2.0',
+    dependencies: {
+      qar: `file:${absolutePath}`,
+    },
+  })
+})

